### ISSUE TITLE
Is there anyone will fix the tsp problem?

### ISSUE THREAD

**Issue by xinwf** [ipa320/autopnp#17]
The tsp algorithm still has problem, will anyone will solve it?

**Comment by AhmedDhahri** [comment]
It is here https://github.com/ipa320/ipa_tsp_solvers
It worked for me

**Comment by KamalanathanN** [comment]
@AhmedDhahri  Did it work for ROS Melodic ?

**Comment by AhmedDhahri** [comment]
@KamalanathanN They have a newer version that works on ROS melodic 
https://github.com/ipa320/ipa_coverage_planning/tree/melodic_dev

**Comment by KamalanathanN** [comment]
@AhmedDhahri Yes i am currently trying to implement the ipa_coverage_planning with the ipa's [cob_robot](https://github.com/ipa320/cob_robots) , inside [cob_simulation](https://github.com/ipa320/cob_simulation)
ROBOT=raw3-3 and ROBOT_ENV=ipa-apartment 
my commands in order are :
```
roslaunch cob_bringup_sim robot.launch
roslaunch cob_navigation_global 2dnav_ros_dwa.launch
roslaunch cob_navigation_global rviz.launch
roslaunch ipa_room_exploration room_exploration_action_server.launch
roslaunch ipa_room_exploration room_exploration_client.launch
```
I get the path generated for ipa-apartment in rviz but its not correctly alinged with the map.
as shown below:
![ipa_room_exploration](https://user-images.githubusercontent.com/31615486/103007239-12845380-4559-11eb-911e-1cf5de784de7.png)

also i have editied the yaml file as mentioned in #https://github.com/ipa320/autopnp/issues/12#issuecomment-495279371
so its supposed to follow the path generated but it's not executing the generated path.
what am i missing here :(

**Comment by eejacky** [comment]
> @KamalanathanN They have a newer version that works on ROS melodic
> https://github.com/ipa320/ipa_coverage_planning/tree/melodic_dev

can you build this version(ipa_coverage_planning-melodic_dev)?
I build with this version ,has some problems.
i build step:
mkdir ~/ros_ws/src
cd ~/ros_ws/src
catkin_init_workspace
cd ~/ros_ws
catkin_make
cd src
git clone https://github.com/ipa320/ipa_coverage_planning.git
git clone https://github.com/ipa320/cob_extern.git
git clone https://github.com/ipa320/cob_perception_common.git

cd ..
catkin_make
executing catkin_make,i get some error:
CMake Error at /opt/ros/melodic/share/catkin/cmake/catkinConfig.cmake:83 (find_package):
  Could not find a package configuration file provided by
  "cob_map_accessibility_analysis" with any of the following names:

    cob_map_accessibility_analysisConfig.cmake
    cob_map_accessibility_analysis-config.cmake

  Add the installation prefix of "cob_map_accessibility_analysis" to
  CMAKE_PREFIX_PATH or set "cob_map_accessibility_analysis_DIR" to a
  directory containing one of the above files.  If
  "cob_map_accessibility_analysis" provides a separate development package or
  SDK, be sure it has been installed.
Call Stack (most recent call first):
  ipa_coverage_planning-melodic_dev/ipa_room_exploration/CMakeLists.txt:33 (find_package)


-- Configuring incomplete, errors occurred!

**Comment by KamalanathanN** [comment]
have you installed cob_map_accessibility_analysis ?
```
sudo apt install ros-melodic-cob-map-accessibility-analysis
```
also try cloning: 
https://github.com/KamalanathanN/ipa_coverage_planning/tree/melodic_dev
i've made small changes and it compiled on melodic.

now im using ros noetic, to test it out in noetic
try cloning @AhmedDhahri's repo:
https://github.com/AhmedDhahri/ipa_coverage_planning
when i remove the uqtr_room_decomposition package it compiles fine.
